### PR TITLE
Notification end point

### PR DESCRIPTION
**Pull request by srinivasupadhya**

### CODE REVIEW

File: plugin-infra/go-plugin-access/src/com/thoughtworks/go/plugin/access/notification/NotificationExtension.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    private Map<String, JsonMessageHandler> messageHandlerMap = new HashMap<String, JsonMessageHandler>();
+
+    @Autowired
+    public NotificationExtension(PluginManager defaultPluginManager) {
+        this.pluginManager = defaultPluginManager;
+        this.pluginRequestHelper = new PluginRequestHelper(pluginManager, goSupportedVersions, EXTENSION_NAME);
+        this.messageHandlerMap.put("1.0", new JsonMessageHandler1_0());
+    }
+
+    public List<String> getNotificationsInterestedIn(String pluginId) {
```
Comment (jyotisingh) [inline plugin-infra/go-plugin-access/src/com/thoughtworks/go/plugin/access/notification/NotificationExtension.java]: method name does not read very well, think about renaming it

File: server/src/com/thoughtworks/go/server/messaging/plugin/PluginNotificationListenerFactory.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        this.systemEnvironment = systemEnvironment;
+        this.notificationExtension = notificationExtension;
+        this.serverHealthService = serverHealthService;
+    }
+
+    public void init() {
+        int numberOfListeners = systemEnvironment.getNumberOfPluginNotificationListener();
+
+        for (int i = 0; i < numberOfListeners; i++) {
+            PluginNotificationService pluginNotificationService = new PluginNotificationService(notificationExtension, serverHealthService);
```
Comment (jyotisingh) [inline server/src/com/thoughtworks/go/server/messaging/plugin/PluginNotificationListenerFactory.java]: PluginNotificationService is a component, why initialize it here again. You could pass it along through the autowired constructor of PluginNotificationListenerFactory.

Comment (srinivasupadhya) [inline server/src/com/thoughtworks/go/server/messaging/plugin/PluginNotificationListenerFactory.java]: on similar lines as [this](https://github.com/gocd/gocd/blob/master/server/src/com/thoughtworks/go/server/materials/MaterialUpdateListenerFactory.java#L77). Need to check why it isn't autowired here as well.

File: plugin-infra/go-plugin-access/src/com/thoughtworks/go/plugin/access/notification/NotificationExtension.java (hunk 1 above)
Comment (srinivasupadhya) [inline plugin-infra/go-plugin-access/src/com/thoughtworks/go/plugin/access/notification/NotificationExtension.java]: ok. will make it `getNotificationsInterestedInFor(pluginId)`

Comment (jyotisingh) [inline plugin-infra/go-plugin-access/src/com/thoughtworks/go/plugin/access/notification/NotificationExtension.java]: Does notificationsOfInterestFor(pluginId) sound about alright?

**Comment by srinivasupadhya** [comment]
done as part of #856

**Pull request merged**